### PR TITLE
ssh_login windows 10/2019 friendly, and 'submit unknown' banner.

### PR DESCRIPTION
**Pull request by h00die**
Part 2 of #13315

This updates `ssh_login` to work with OpenSSH on Windows as per https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse
It also adds that if you `gatherproof` (default) and the session comes back as `unknown`, to submit an issue so we can get it added.

### CODE REVIEW

**Comment by h00die** [comment]
With patch and a !'unknown' platform:
```
msf5 auxiliary(scanner/ssh/ssh_login) > rexploit
[*] Reloading module...

[+] 2.2.2.2:22 - Success: 'user:password' 'Microsoft Windows 10 Pro 10.0.18363 N/A Build 18363'
[*] Command shell session 3 opened (1.1.1.1:44143 -> 2.2.2.2:22) at 2020-04-23 17:28:02 -0400
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```
I undid the Windows detection code to show what the results would be for an 'unknown' platform.
```
msf5 auxiliary(scanner/ssh/ssh_login) > run

[+] 2.2.2.2:22 - Success: 'user:password' ''id' is not recognized as an internal or external command,  operable program or batch file.  '
[*] Command shell session 4 opened (1.1.1.1:44561 -> 2.2.2.2:22) at 2020-04-23 17:29:03 -0400
[-] 2.2.2.2:22 - While a session may have opened, it may be bugged.  If you experience issues with it, re-run this module with 'set gatherproof off'.  Also consider submitting an issue at github.com/rapid7/metasploit-framework with device details so it can be handled in the future.
[*] Scanned 1 of 1 hosts (100% complete)
```

It is a bit wordy, up for suggestions.

File: lib/metasploit/framework/login_scanner/ssh.rb [hunk 1]
```diff
@@ -152,6 +152,14 @@ def gather_proof
                   if proof =~ /Version:(?<os_version>.+).+HW: (?<hardware>)/mi
                     proof = "Model: #{hardware}, OS: #{os_version}"
                   end
+                # Windows
+                elsif proof =~ /is not recognized as an internal or external command/
+                  proof = ssh_socket.exec!("systeminfo\n").to_s
+                  /OS Name:\s+(?<os_name>.+)$/ =~ proof
```
Comment (wvu-r7) [inline lib/metasploit/framework/login_scanner/ssh.rb:158]: These Perl-style regexes are a little weird on their own lines, but the whole file's strange. Leaving it alone this time...

File: modules/auxiliary/scanner/ssh/ssh_login.rb [hunk 2]
```diff
@@ -132,6 +132,12 @@ def run_host(ip)
         credential_data[:core] = credential_core
         create_credential_login(credential_data)
         session_setup(result, scanner) if datastore['CreateSession']
+        if datastore['GatherProof'] && scanner.get_platform(result.proof) == 'unknown'
+          msg = "While a session may have opened, it may be bugged.  If you experience issues with it, re-run this module with"
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_login.rb:136]: I'm gonna chalk this message up to being a @h00die-ism. :)

Comment (wvu-r7) [inline modules/auxiliary/scanner/ssh/ssh_login.rb:136]: I'm gonna chalk this message up to being a h00die-ism. :)

Comment (wvu-r7) [reply modules/auxiliary/scanner/ssh/ssh_login.rb:136]: Also, this will print for every IP with unknown platform when `GatherProof` is set (new default).

Comment (wvu-r7) [reply modules/auxiliary/scanner/ssh/ssh_login.rb:136]: We'll see how much we like it. Don't care that much right now. `¯\_(ツ)_/¯`

**Comment by wvu-r7** [comment]
# Release Notes

This enhances the `GatherProof` advanced option in SSH login scanners to handle Windows and unknown platforms better.
